js/geo/transform.js

```diff
@@ -453,6 +453,10 @@ class Transform {
         mat4.rotateX(m, m, this._pitch);
         mat4.rotateZ(m, m, this.angle);
         mat4.translate(m, m, [-this.x, -this.y, 0]);
+        mat4.scale(m, m, [1, 1,
+            // scale vertically to meters per pixel (inverse of ground resolution)
+            (Math.pow(2, this.zoom) * 512) / (2 * Math.PI * 6378137 * Math.abs(Math.cos(this.center.lat * (Math.PI / 180)))),
```

Comment:
Can you document or split this up a bit more? maybe replace `2 * Math.PI * 6378137` with a `circumferenceOfEarth` constant